common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -246,6 +304,14 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
     }
   }
 
+  private void failSingleBlockChunk(String shuffleBlockChunkId, Throwable e) {
+    try {
+      listener.onBlockFetchFailure(shuffleBlockChunkId, e);
+    } catch (Exception e2) {
+      logger.error("Error from blockFetchFailure callback", e2);
+    }
+  }
```

Comment:
We can have `failRemainingBlocks` delegate to `failSingleBlockChunk` now ?
```
  private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
    Arrays.stream(failedBlockIds).forEach(blockId -> failSingleBlockChunk(blockId, e));
  }
```

Reply:
Thanks for bringing this up @mridulm. This is something I missed as the push-based shuffle code evolved. We don't need this change and I will revert it.
We do want to fail all the remaining chunkIds as well so that they get retried by `RetryingBlockFetcher`. 
Even with this code, the remaining chunkIds are going to be retried by `RetryingBlockFetcher` because that not only retries the fetch of the failed block but also the fetch of all the outstanding blocks.  In addition `RetryingBlockFetchListener.initiateRetry` also changes the currentListener so any invocation of `onBlockFetchFailure()` on an old instance of `RetryingBlockFetchListener` would do nothing because that checks whether the instance is the current on.

In short, this change is unnecessary here and is not needed.